main/src/main/java/com/sedmelluq/discord/lavaplayer/source/nico/NicoAudioTrack.java

```diff
@@ -84,31 +84,32 @@ private JSONObject loadVideoMainPage(HttpInterface httpInterface) throws IOExcep
                 throw new IOException("Unexpected status code from video main page: " + statusCode);
             }
 
-            return (JSONObject) new JSONObject(
-                Jsoup.parse(response.getEntity().getContent(), StandardCharsets.UTF_8.name(), "")
-                    .getElementById("js-initial-watch-data").attributes().get("data-api-data"))
-                .query("/media/delivery/movie/session");
+            Document mainPage = Jsoup.parse(response.getEntity().getContent(), StandardCharsets.UTF_8.name(), "");
+            String watchdata = mainPage.getElementById("js-initial-watch-data").attributes().get("data-api-data");
+
+            return JsonBrowser.parse(watchdata).get("media").get("delivery").get("movie").get("session");
         }
     }
 
     private String loadPlaybackUrl(HttpInterface httpInterface) throws IOException {
+        JsonBrowser watchdata = processJSON(loadVideoMainPage(httpInterface));
         HttpPost request = new HttpPost("https://api.dmc.nico/api/sessions?_format=json");
         request.addHeader("Host", "api.dmc.nico");
         request.addHeader("Connection","keep-alive");
         request.addHeader("Content-Type","application/json");
         request.addHeader("Origin","https://www.nicovideo.jp");
-        request.setEntity(new StringEntity(processJSON(loadVideoMainPage(httpInterface)).toString()));
+        request.setEntity(new StringEntity(watchdata.text()));
```

Comment:
I don't think `.text()` does what you think it does here. You're perhaps looking for `.format()`, which converts the entire JSON node and subnodes into a string.

Although I suppose I should've been clearer in my original review - we use `JsonBrowser` for deserialization, and `org.json` for serialization, as `JsonBrowser` isn't *really* geared towards serialization.